library/src/main/java/com/getstream/sdk/chat/adapter/CommandsAdapter.kt

```diff
@@ -0,0 +1,50 @@
+package com.getstream.sdk.chat.adapter
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.databinding.DataBindingUtil
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.getstream.sdk.chat.R
+import com.getstream.sdk.chat.databinding.StreamItemCommandBinding
+import com.getstream.sdk.chat.databinding.StreamItemMentionBinding
+import com.getstream.sdk.chat.view.BaseStyle
+import com.getstream.sdk.chat.view.MessageInputStyle
+import io.getstream.chat.android.client.models.Command
+
+class CommandsAdapter(
+		private val style: BaseStyle,
+		private val onCommandSelected: (Command) -> Unit
+) : ListAdapter<Command, CommandViewHolder>(
+		object : DiffUtil.ItemCallback<Command>() {
+			override fun areItemsTheSame(oldItem: Command, newItem: Command): Boolean = oldItem.name == newItem.name
+			override fun areContentsTheSame(oldItem: Command, newItem: Command): Boolean = oldItem == newItem
+		}
+) {
+	override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CommandViewHolder =
+			CommandViewHolder(
+					StreamItemCommandBinding.inflate(LayoutInflater.from(parent.context), parent, false),
+					style,
+					onCommandSelected
+			)
+	override fun onBindViewHolder(holder: CommandViewHolder, position: Int) = holder.bind(getItem(position))
+}
+
+class CommandViewHolder(
+		private val binding: StreamItemCommandBinding,
+		private val style: BaseStyle,
```

Comment:
`BaseStyle` by the way also should be eventually refactored. Currently it has lots of particular (not base at all) properties like `readStateAvatarWidth`. Not sure if base style is even needed.

Reply:
I tried to refactor it when I started to work on `AvatarView` but it was no easy, I achieve to remove the unneeded generic type it was using, but BaseStyle is being used in lot of places, so it will be refactored in future tasks. I think we can simplify it a lot, and try to do all configuration with themes, styles, and custom attributes.

Right now I am only following the previous implementation, keeping it as simple as possible